crates/typst-cli/src/fonts.rs

```diff
@@ -1,20 +1,23 @@
+use crate::args::FontsCommand;
+use std::io::Write;
 use typst::text::FontVariant;
 use typst_kit::fonts::Fonts;
-
-use crate::args::FontsCommand;
-
 /// Execute a font listing command.
 pub fn fonts(command: &FontsCommand) {
     let fonts = Fonts::searcher()
         .include_system_fonts(!command.font_args.ignore_system_fonts)
         .search_with(&command.font_args.font_paths);
-
+    let mut out = crate::terminal::out();
     for (name, infos) in fonts.book.families() {
-        println!("{name}");
+        writeln!(out, "{name}:").unwrap();
         if command.variants {
             for info in infos {
                 let FontVariant { style, weight, stretch } = info.variant;
-                println!("- Style: {style:?}, Weight: {weight:?}, Stretch: {stretch:?}");
+                writeln!(
+                    out,
+                    "- Style: {style:?}, Weight: {weight:?}, Stretch: {stretch:?}"
+                )
+                .unwrap();
```

Comment:
It is tested that now it works well.

Reply:
> I think unwrapping here might generate the same problem, but either way we should at least make `fonts` fallible and return an IO error instead of crashing.

https://github.com/typst/typst/pull/5444#issuecomment-2492409056

Reply:
I apologize if that was confusing: I'm not proposing to ignore the error, I just think it would be more consistent with other commands if we made `fn fonts()` return a `Result` (say, with `io::Error` as the error type) and then used `writeln!(out, ...)?`.

Reply:
Now, query() would return a Result, but it will just `unwarp` the error given in writeln. All other functions will give a hint on this error, but I am not sure what hint we could give if we have an error in write.

Reply:
I'm thinking of completely removing unwrap from usages of `writeln!` and doing similarly to `print_diagnostics` as follows:

https://github.com/typst/typst/blob/5c37a1cfea2a08b08f4af34ba867748fd90c687e/crates/typst-cli/src/query.rs#L31-L33

A similar `.map_err()?` should be enough.